kong/llm/drivers/shared.lua

```diff
@@ -229,6 +242,75 @@ local function handle_stream_event(event_table, model_info, route_type)
   end
 end
 
+---
+-- Manages cloud SDKs, for using "workload identity" authentications,
+-- that are tied to this specific plugin in-memory.
+--
+-- This allows users to run different authentication configurations
+-- between different AI Plugins.
+--
+-- @param {table} this_cache self - stores all the SDK instances
+-- @param {table} plugin_config the configuration to cache against and also provide SDK settings with
+-- @return {table} self
+_M.cloud_identity_function = function(this_cache, plugin_config)
+  if plugin_config.model.provider == "gemini" and
+      plugin_config.auth and
+      plugin_config.auth.gcp_use_service_account then
+
+    ngx.log(ngx.NOTICE, "loading gcp sdk for plugin ", kong.plugin.get_id())
```

Comment:
Do we really need NOTICE level log here? This could be a little bit noisy.

Reply:
This only logs e.g. once every hour, after tokens refresh, but I've changed it anyway to make less noise.